lib/authorized_persona/authorization.rb

```diff
@@ -41,21 +41,21 @@ def grant(privileges) # rubocop:disable Metrics/AbcSize
         tier_names = authorization_persona.authorization_tier_names
         extra_keys = authorized_actions.keys - authorization_persona.authorization_tier_names
         if extra_keys.present?
-          raise AuthorizedPersona::Error, "invalid grant: #{authorization_persona_class_name} " \
-                                          "has authorization tiers #{tier_names.join(', ')} but received extra keys: #{extra_keys.join(', ')}"
+          raise Error, "invalid grant: #{authorization_persona_class_name} " \
+                       "has authorization tiers #{tier_names.join(', ')} but received extra keys: #{extra_keys.join(', ')}"
```

Comment:
__question:__ Are the order of the entries in `tier_names` and `extra_keys` deterministic?

Reply:
yes, tier_names are a result of calling hash.keys (resulting in an array) on the tier definition hash (which cares about order because the sequence defines the privilege ladder). extra_keys is then the result of subtracting an array from an array, which yields an array in order, so deterministic, yup.

Reply:
@dkubb is there an OrderedHash or equivalent concept in std lib? i am not aware of such a thing, but it would be preferred. ultimately we could just have our own structure that internally uses a pair of arrays.

Reply:
it would be easier to add a test to just ensure that if hash stops being ordered then the test fails 😄

Reply:
> is there an OrderedHash or equivalent concept in std lib

__comment:__ In modern versions of ruby the Hash insertion order is preserved and exposed via `#keys` and others.

I think I was mostly curious if the inputs that were used to create the Hash are deterministically ordered. Based on the [above comment](https://github.com/Betterment/authorized_persona/pull/8/files#r1388062829) sounds like this is the case. If they were nondeterministic, then we'd see occasional flakes, which was my primary concern.